warehouse/redshift_test.go

```diff
@@ -36,3 +36,44 @@ func TestRedshiftValueToString(t *testing.T) {
 		}
 	}
 }
+
+func TestGetMissingFields(t *testing.T) {
+	wh := &Redshift{
+		exportSchema: ExportTableSchema(RedshiftTypeMap),
+	}
+
+	schemaHeaders := wh.getSchemaHeaders(wh.exportSchema)
+	var noHeaders []string
+	var testCases = []struct {
+		schema    Schema
+		columns   []string
+		expected  int
+	} {
+		// All columns from the schema are present in the export table columns, so there are 0 missing fields
+		{wh.exportSchema, schemaHeaders, 0},
+		// Only headers are present, therefore all schema fields are missing
+		{wh.exportSchema, noHeaders, len(schemaHeaders)},
+		// Only headers that are not part of the schema are present, therefore all schema fields are missing
+		{wh.exportSchema, []string{"Dummy"}, len(schemaHeaders)},
+		// Only one column common between the table columns, and schema, so there are len(schemaHeaders) - 1 missing field
+		{wh.exportSchema, []string{"PageUrl"}, len(schemaHeaders) - 1},
+		// Same as above, but with additional columns in the export table that we don't care about so still 1 missing field
+		{wh.exportSchema, []string{"Dummy Column1", "PageUrl", "Dummy Column2"}, len(schemaHeaders) - 1},
+
+	}
+
+	for _, testCase := range testCases {
+		missingFieldNames := getFieldNames(wh.getMissingFields(testCase.schema, testCase.columns))
```

Comment:
I don't think we have to call `getFieldNames` here, as the test only ever checks the length of the slice? Or were you planning on adding more checks explicitly on the field name values?

Reply:
I think you have enough cases that this test is ok. If we were to check field names, the test would be brittle, because every time the schema changed, the test would be likely to break. 

I think if I were to build the test from scratch, I would construct a test schema with only enough fields to cover the test cases. Then checking the names of each field in the result wouldn't be a huge task, and you could likely combine a couple of the test cases.

Reply:
for this PR I will stop at checking just the lengths. In a separate PR, will work on adding the ability to build custom schemas and check field names in the tests.